src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/Interfaces/ITestSession.cs

```diff
@@ -151,30 +169,42 @@ public interface ITestSession : ITestSessionAsync
         /// <param name="options">The test platform options.</param>
         /// <param name="testRunEventsHandler">The run event handler.</param>
         /// <param name="customTestHostLauncher">The custom host launcher.</param>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
         void RunTestsWithCustomTestHost(
             IEnumerable<TestCase> testCases,
             string runSettings,
             TestPlatformOptions options,
             ITestRunEventsHandler testRunEventsHandler,
             ITestHostLauncher customTestHostLauncher);
 
+        /// <summary>
+        /// Stops the test session.
+        /// </summary>
+        /// 
+        /// <returns>True if the session was successfuly stopped, false otherwise.</returns>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
+        bool StopTestSession();
+
         /// <summary>
         /// Stops the test session.
         /// </summary>
         /// 
         /// <param name="eventsHandler">The session event handler.</param>
         /// 
         /// <returns>True if the session was successfuly stopped, false otherwise.</returns>
+        [Obsolete("This API is not final yet and is subject to changes.", false)]
         bool StopTestSession(ITestSessionEventsHandler eventsHandler);
```

Comment:
if the StartTestSession API takes an EventsHandler, does StopTestSession also need to take an events handler?

Reply:
It does, because it's the only way of letting the TE know if the `StopTestSession` call was successful or not. The details of what means an unsuccessful call are in one of the replies above, but in short an unsuccessful call doesn't mean that some testhosts are left running, but rather that `StopTestSession` has already been call for the same test session.